tests/cli/test_remote.py

```diff
@@ -1,7 +1,731 @@
+    def test_srun_transform(self, mock_connection: Connection):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        persist: bool = False
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        # Transforms aren't used here. Seems a bit weird for this to be a public method
+        # then, no?
+        assert remote.srun_transform("bob") == "srun --time=00:01:00 bash -c bob"
+
+    @dont_run_for_real
+    @pytest.mark.skip(reason="Seems a bit hard to test for what it's worth..")
+    def test_srun_transform_persist(self, mock_connection: Connection):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        persist: bool = False
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        output_file = "<some_file>"
+        assert (
+            remote.srun_transform_persist("bob")
+            == f"bob; touch {output_file}; tail -n +1 -f {output_file}"
+        )
+
+    @dont_run_for_real
+    @pytest.mark.parametrize("persist", [True, False, None])
+    def test_with_transforms(self, mock_connection: Connection, persist: bool | None):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        original_persist: bool = False
+        remote = SlurmRemote(
+            mock_connection,
+            alloc=alloc,
+            transforms=transforms,
+            persist=original_persist,
+        )
+        new_transforms = [some_other_transform]
+        transformed = remote.with_transforms(*new_transforms, persist=persist)
+        # NOTE: Feels dumb to do this. Not sure what I should be doing otherwise.
+        assert transformed.connection == remote.connection
+        assert transformed.alloc == remote.alloc
+        assert transformed.transforms == [
+            some_transform,
+            some_other_transform,
+            (
+                transformed.srun_transform_persist
+                if persist
+                else transformed.srun_transform
+            ),
+        ]
+        assert transformed._persist == (remote._persist if persist is None else persist)
+
+    @dont_run_for_real
+    @pytest.mark.parametrize("persist", [True, False])
+    def test_persist(self, mock_connection: Connection, persist: bool):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        persisted = remote.persist()
+
+        # NOTE: Feels dumb to do this. Not sure what I should be doing otherwise.
+        assert persisted.connection == remote.connection
+        assert persisted.alloc == remote.alloc
+        assert persisted.transforms == [
+            some_transform,
+            persisted.srun_transform_persist,
+        ]
+        assert persisted._persist is True
```

Comment:
Same, I think we could mock and use a `FileRegressionFixture` or something like that. And maybe have a test with a real connection to a slurm cluster